src/components/AssessmentView.jsx

```diff
@@ -201,34 +206,45 @@ const AssessmentView = ({
       },
     ];
   };
-
   const handleScoreOnchange = (e, students, weightage) => {
-    const updatedScore = {
-      task_user: students.id,
-      task_weightage: weightage.id,
-      task_score: Number(e.target.value),
-    };
-    // Check if the score for the same student and weightage ID already exists
-    const existingScoreIndex = studentScore.findIndex(
-      (score) =>
-        score.task_user === updatedScore.task_user &&
-        score.task_weightage === updatedScore.task_weightage
-    );
-
-    if (existingScoreIndex !== -1) {
-      // If the score exists, update it
-      const updatedStudentScores = [...studentScore];
-      updatedStudentScores[existingScoreIndex].task_score =
-        updatedScore.task_score;
-      setStudentScore(updatedStudentScores);
+    const scoreValue = Number(e.target.value);
+  
+    // Check if the score is 0, and if so, remove the corresponding object from the state
+    if (scoreValue === 0) {
```

Comment:
what if the admin puts zero as score mark, on that case this will fail, we need to discuss on this

Reply:
We will discuss